### PR TITLE
ensure optimize-relinearization variable names are unique

### PR DESCRIPTION
**Pull request by j2kun**
Fixes https://github.com/google/heir/issues/1148

### CODE REVIEW

File: lib/Analysis/OptimizeRelinearizationAnalysis/OptimizeRelinearizationAnalysis.cpp [hunk 1]
```diff
@@ -67,6 +61,32 @@ bool hasCiphertextType(ValueRange range) {
...
+  int resultFreeVarNameIndex = 0;
+  auto uniqueName = [&](Operation *op) {
+    std::string varName;
+    llvm::raw_string_ostream ss(varName);
+    ss << op->getName() << "_";
+    if (useLocBasedVariableNames) {
+      ss << op->getLoc();
+    } else if (op->getNumResults() == 0) {
+      ss << resultFreeVarNameIndex++;
+    } else {
```
Comment (asraa) [inline lib/Analysis/OptimizeRelinearizationAnalysis/OptimizeRelinearizationAnalysis.cpp:84]: QQ - why not only use the free var name index if you have to use it for the 0 result case? Can you just omit SelectVariableNames in this pass completely if you already have a free var name counter?
(I guess for simplicity why not apply this to all ops, regardless of how many results they have).

**Review by asraa — COMMENTED** [review]
Nice fix!

File: lib/Analysis/OptimizeRelinearizationAnalysis/OptimizeRelinearizationAnalysis.cpp (hunk 1 above)
Comment (j2kun) [reply lib/Analysis/OptimizeRelinearizationAnalysis/OptimizeRelinearizationAnalysis.cpp:84]: Yeah, I could just give up and use completely opaque identifiers when the pass option is not set. I think I will do that.

**Review by asraa — APPROVED** [review]

**Review by AlexanderViand-Intel — APPROVED** [review]
LGTM

**Pull request merged**